comps/dataprep/src/integrations/elasticsearch.py

```diff
@@ -142,7 +142,7 @@ def search_by_filename(self, file_name: str) -> bool:
 
         return results["hits"]["total"]["value"] > 0
 
-    def ingest_doc_to_elastic(self, doc_path: DocPath) -> None:
+    async def ingest_doc_to_elastic(self, doc_path: DocPath) -> None:
```

Comment:
General question, if we are given a list of documents, can we process/injest them in parallel? Further, could one split a document into chunks and process the chunks in parallel? Would certainly need load balancing over an embedding service and serialize at the point of inserting in the vector database and use write locks etc.

Reply:
Good idea. This may need owner @letonghan  for an suggestion~